src/quo/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,218 @@
+(ns quo.components.wallet.transaction-progress.view
+  (:require [quo.components.buttons.button.view :as button]
+            [quo.components.icon :as icons]
+            [quo.components.markdown.text :as text]
+            [quo.components.tags.context-tag.view :as context-tag]
+            [quo.components.wallet.progress-bar.view :as progress-box]
+            [quo.components.wallet.transaction-progress.style :as style]
+            [quo.foundations.colors :as colors]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]
+            [utils.i18n :as i18n]))
+
+(defn icon-internal
+  ([icon]
+   (icon-internal icon nil 20))
+  ([icon color]
+   (icon-internal icon color 20))
+  ([icon color size]
+   [rn/view {:style style/icon}
+    [icons/icon
+     icon
+     {:color    color
+      :size     size
+      :no-color (when (not color) true)}]]))
+
+(defn calculate-box-state
+  [state counter index]
+  (cond
+    (and (= state :sending) (>= counter index) (< index 3))                 :confirmed
+    (and (= state :confirmed) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalising) (>= counter index) (< index 5))              :confirmed
+    (and (= state :finalising) (>= counter index) (> index 4) (< index 20)) :finalized
+    (and (= state :finalized) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalized) (>= counter index) (> index 4))               :finalized
+    (and (= state :error) (>= counter index) (< index 2))                   :error
+    :else                                                                   :pending))
+
+(defn progress-boxes
+  [state counter total-box]
+  [rn/view
+   {:accessibility-label :mainnet-progress-box
+    :style               (style/progress-box-container true)}
+   (let [numbers (range 1 total-box)]
+     (doall (for [n numbers]
+              [progress-box/view
+               {:state               (calculate-box-state state counter n)
+                :customization-color :blue
+                :key                 n}])))])
+
+(defn calculate-box-state-arbitrum-left
+  [state network]
+  (cond
+    (= state :error)                                                     :error
+    (and (= network :arbitrum) (= state :sending))                       :confirmed
+    (or (= state :confirmed) (= state :finalising) (= state :finalized)) :confirmed
+    :else                                                                :pending))
+
+(defn calculate-box-state-arbitrum-right
+  [state network]
+  (cond
+    (= state :error)
+    :error
+    (and (= network :arbitrum)
+         (= state :sending))
+    :confirmed
+    (or (= state :confirmed)
+        (= state :finalising)
+        (= state :finalized))
+    :finalized
+    :else
+    :pending))
+
+(defn calculate-progressed-value
+  [state network]
+  (case state
+    :finalising (if (= network :arbitrum) 26 16)
```

Comment:
Could you please elaborate a bit, what these magic numbers mean?

Reply:
I will change the codes and put them in consts for better readability

Reply:
Done